mesonbuild/interpreter/interpreter.py

```diff
@@ -2753,12 +2753,24 @@ def extract_incdirs(self, kwargs, key: str = 'include_directories'):
         return result
 
     @typed_pos_args('include_directories', varargs=str)
-    @typed_kwargs('include_directories', KwargInfo('is_system', bool, default=False))
+    @typed_kwargs(
+        'include_directories',
+        KwargInfo('is_system', bool, default=False),
+        KwargInfo(
+            'use_type',
+            str,
+            default='src-and-build-relative',
```

Comment:
these names are also rather long and verbose `src`, `build`, `both` are much shorter and easier to remember.

I'm also not sure about the name. `use_type` isn't wrong, but maybe we can come up with something that is more self-descriptive?

Reply:
Fine with `both`, `build`, `src`.

I also wasn't sure about `use_type` but couldn't come up with anything better and barring something really long and descriptive, which is itself still quite difficult to encapsulate the meaning of this field, I couldn't think of anything better.  If either of you can, I'm all ears 😄